test/test_luaunit.lua

```diff
@@ -1957,6 +1960,7 @@ TestLuaUnitAssertionsError = {}
         assertFailure( lu.assertErrorMsgMatches, 'is an err', self.f_with_error, x )
         lu.assertErrorMsgMatches( 'This is an error', self.f_with_error, x )
         lu.assertErrorMsgMatches( 'This is .. error', self.f_with_error, x )
+        lu.assertErrorMsgMatches("^500$", function() error(500) end)
```

Comment:
try with error(500, 2). I expect that this would fix the problem for lua 5.1 and 5.2 . At least, this is the only difference that I could spot versus self.f_with_error()